packages/babel-helpers/src/helpers/interopRequireWildcard.ts

```diff
@@ -1,54 +1,48 @@
 /* @minVersion 7.14.0 */
 
-function _getRequireWildcardCache(nodeInterop: boolean) {
-  if (typeof WeakMap !== "function") return null;
-
-  var cacheBabelInterop = new WeakMap();
-  var cacheNodeInterop = new WeakMap();
-  // @ts-expect-error assign to function
-  return (_getRequireWildcardCache = function (nodeInterop: boolean) {
-    return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
-  })(nodeInterop);
-}
-
 export default function _interopRequireWildcard(
   obj: any,
   nodeInterop: boolean,
 ) {
-  if (!nodeInterop && obj && obj.__esModule) {
-    return obj;
+  if (typeof WeakMap === "function") {
+    var cacheBabelInterop = new WeakMap();
+    var cacheNodeInterop = new WeakMap();
   }
 
-  if (obj === null || (typeof obj !== "object" && typeof obj !== "function")) {
-    return { default: obj };
-  }
+  // @ts-expect-error: assign to function
+  return (_interopRequireWildcard = function (obj: any, nodeInterop: boolean) {
+    if (!nodeInterop && obj && obj.__esModule) {
+      return obj;
+    }
+    // Temporary variable for output size
+    var _: any = nodeInterop ? cacheNodeInterop : cacheBabelInterop;
```

Comment:
```suggestion
    var _;
    _ = nodeInterop ? cacheNodeInterop : cacheBabelInterop;
```

If we don't explicitly annotate as `any`, TS will dynamically type it based on the last assignment (so it will first be typed as `WeakMap`, and then as `typeof Object.defineProperty`).